mm2src/lp_ordermatch.rs

```diff
@@ -2126,7 +2126,7 @@ impl Orderbook {
                 return;
             },
         };
-        let order_bytes = rmp_serde::to_vec(&order).expect("Serialization should never fail");
+        let order_bytes = rmp_serde::to_vec(&order.remove_protocol_info()).expect("Serialization should never fail");
```

Comment:
Correct me please if I'm wrong. I think there can be a not bad but not good case if we don't store the base/rel protocol info in a `memory_db`.

Let's there are three nodes in a network: our light node `A`, another light node `B`, seed node `S`.

1) `B` broadcasts a few Maker orders to the network. `A` and `S` successfully sync them by [process_keep_alive](https://github.com/KomodoPlatform/atomicDEX-API/blob/dev/mm2src/lp_ordermatch.rs#L2281).
2) `A` goes offline.
3) `B` broadcasts a new Maker order, `S` syncs it, therefore stores the order in `Ordermatch::memory_db` on [process_trie_delta](https://github.com/KomodoPlatform/atomicDEX-API/blob/dev/mm2src/lp_ordermatch.rs#L148). Note that the order is stored without protocol info since we use `rmp_serde::to_vec(&order.remove_protocol_info())`.
4) `A` connects to the `S` and requests the diff of the orderbook on [process_orders_keep_alive](https://github.com/KomodoPlatform/atomicDEX-API/blob/dev/mm2src/lp_ordermatch.rs#L187) from `S` node
5) `S` [extracts](https://github.com/KomodoPlatform/atomicDEX-API/blob/dev/mm2src/lp_ordermatch.rs#L707) the missed order without protocol info and send it to `A`
6) `A` generates and broadcasts a Taker order to match the `B`'s Maker order BUT without the protocol info.
7) I guess, `B`doesn't match the Taker order since there are no base/rel protocol infos (not sure about it)

Reply:
I think it will match because in point no. 6 `A` will generate and broadcast a Taker order with its own protocol info and on the maker side `B` will check if it can match with `A` or not. So `A` doesn't use protocol info. of `B` in `TakerRequest`, `A` only sends its own protocol info to let `B` know if it's an old node when protocol info of `A` is None thus not matching `B` Segwit orders only in this case.
I will try to add a test case for the above scenario to check it but it might be tricky disconnecting and connecting `A` again in testing.

Reply:
Have you tried to reproduce this? If you are busy now, I can do this)

Reply:
As we discussed with @shamardy, it may be difficult to reproduce the case above, because the `A` node should be disconnected without restarting it.
@cipig @tonymorony could you please try to reproduce this case?